ui/ui-frontend/projects/pastis/src/app/profile/edit-profile/file-tree-metadata/file-tree-metadata.component.ts

```diff
@@ -0,0 +1,693 @@
+import {FileTreeMetadataService} from './file-tree-metadata.service';
+
+
+const FILE_TREE_METADATA_TRANSLATE_PATH = 'PROFILE.EDIT_PROFILE.FILE_TREE_METADATA';
+
+function constantToTranslate() {
+  this.notificationAjoutMetadonnee = this.translated('.NOTIFICATION_AJOUT_METADONNEE');
+  this.boutonAjoutMetadonnee = this.translated('.BOUTON_AJOUT_METADONNEE');
+  this.boutonAjoutUA = this.translated('.BOUTON_AJOUT_UA');
+  this.popupSousTitre = this.translated('.POPUP_SOUS_TITRE');
+  this.popupValider = this.translated('.POPUP_VALIDER');
+  this.popupAnnuler = this.translated('.POPUP_ANNULER');
+}
+
+@Component({
+  selector: 'pastis-file-tree-metadata',
+  templateUrl: './file-tree-metadata.component.html',
+  styleUrls: ['./file-tree-metadata.component.scss'],
+  // Encapsulation has to be disabled in order for the
+  // component style to apply to the select panel.
+  encapsulation: ViewEncapsulation.None,
+})
+
+export class FileTreeMetadataComponent {
+
+  valueOrData = Object.values(ValueOrDataConstants);
+  dataType = Object.values(DataTypeConstants);
+  cardinalityList: string[];
+  cardinalityLabels = Object.values(CardinalityConstants);
+  selected = -1;
+
+  // Mat table
+  matDataSource: MatTableDataSource<MetadataHeaders>;
+
+  @ViewChild('autosize', {static: false}) autosize: CdkTextareaAutosize;
+
+  displayedColumns: string[] = ['nomDuChamp', 'valeurFixe', 'cardinalite', 'commentaire', 'menuoption'];
+
+  clickedNode: FileNode = {} as FileNode;
+
+  sedaData: SedaData = {} as SedaData;
+
+  // The seda node that has been opened from the left menu
+  selectedSedaNode: SedaData;
+
+  selectedCardinalities: string[];
+
+  allowedSedaCardinalityList: string[][];
+
+  cardinalityValues: CardinalityValues[] = [];
+
+  regexPattern = '';
+
+  patternType: string;
+
+  rowIndex: number;
+
+  hoveredElementId: number;
+
+  buttonIsClicked: boolean;
+
+  isStandalone: boolean = environment.standalone;
+
+  public breadcrumbDataTop: Array<BreadcrumbDataTop>;
+  public breadcrumbDataMetadata: Array<BreadcrumbDataMetadata>;
+
+  profileModeLabel: string;
+
+  config: {};
+
+
+  notificationAjoutMetadonnee: string;
+  boutonAjoutMetadonnee: string;
+  boutonAjoutUA: string;
+  popupSousTitre: string;
+  popupValider: string;
+  popupAnnuler: string;
+
+  @Output()
+  public insertItem: EventEmitter<FileNodeInsertParams> = new EventEmitter<FileNodeInsertParams>();
+
+  @Output()
+  public addNode: EventEmitter<FileNode> = new EventEmitter<FileNode>();
+
+  @Output()
+  public insertAttributes: EventEmitter<FileNodeInsertAttributeParams> = new EventEmitter<FileNodeInsertAttributeParams>();
+
+  @Output()
+  public removeNode: EventEmitter<FileNode> = new EventEmitter<FileNode>();
+
+  private _profileServiceProfileModeSubscription: Subscription;
+
+  @Output()
+  public duplicateNode: EventEmitter<FileNode> = new EventEmitter<FileNode>();
+
+  private _fileServiceSubscription: Subscription;
```

Comment:
Même remarque que plus haut pour gérer les subscriptions : https://dev.to/this-is-angular/dry-way-to-manage-subscriptions-in-angular-components-256j